bundle/edu.gemini.pot/src/main/java/edu/gemini/pot/sp/memImpl/MemTemplateFolder.java

```diff
@@ -86,6 +86,7 @@ public void setTemplateGroups(List<? extends ISPTemplateGroup> newGroupList) thr
         List<ISPTemplateGroup> newCopy = new ArrayList<>(newGroupList);
         getProgramWriteLock();
         try {
+            SPAssert.setsNoDuplicateObs(this, newCopy);
```

Comment:
If the assertion throws an exception it represents a logic error.  It isn't supposed to be possible for this to happen. I believe the OT will catch exceptions at a high level and dump them into a dialog box, even if they are logic errors.

There's nothing the user can really do to fix the problem, though each time you create a new observation the counter is bumped so they will get a new observation number.  If you keep at it long enough, it will eventually get to a new unused observation number that can be added.

Nevertheless, I believe we should keep the assertion because the alternative, allowing a program with duplicate observation numbers to exist because of a logic error, can have worse consequences.